website/docs/docs/collaborate/govern/model-versions.md

```diff
@@ -60,51 +104,140 @@ models:
+        config:
+          materialized: table
+          contract: {enforced: true}
+        columns:
+          - name: customer_id
+            description: This is the primary key
+            data_type: int
+          - name: country_name
+            description: Where this customer lives
+            data_type: varchar
 ```
 
 </File>
 
+</TabItem>
+
+</Tabs>
+
 The above configuration will create two models (one for each version), and produce database relations with aliases `dim_customers_v1` and `dim_customers_v2`.
 
-By convention, dbt will expect those two models to be defined in files named `dim_customers_v1.sql` and `dim_customers_v2.sql`. It is possible to override this by setting a `defined_in` property.
+By convention, dbt will expect those two models to be defined in files named `dim_customers_v1.sql` and `dim_customers_v2.sql`. (It is possible to override this by setting `defined_in: any_file_name_you_want`, but we strongly encourage you to follow the convention!)
+
+The `latest_version` would be `2` (numerically greatest) if not specified explicitly. In this case, `v1` is specified to still be the latest; `v2` is just a prerelease in early development. When ready to roll out `v2` to everyone by default, bump the `latest_version` to `2` (or remove it from the specification).
+
+### Configuring versioned models
 
-You can reconfigure each version independently. For example, if you wanted `dim_customers.v1` to continue populating the database table named `dim_customers` (its original name), you could use the `defined_in` configuration:
+You can reconfigure each version independently. For example, you could materialize `v2` as a table and `v1` as a view:
+
+<File name="models/schema.yml">
+
+```yml
+versions:
+  - v: 2
+    config:
+      materialized: table
+  - v: 1
+    config:
+      materialized: view
+```
+
+</File>
+
+Like with all config inheritance, any configs set _within_ the versioned model's definition (`.sql` or `.py` file) will take precedence over the configs set in yaml.
+
+### Configuring database location with `alias`
+
+Following the example, let's say you wanted `dim_customers.v1` to continue populating the database table named `dim_customers`. That's what the table was named previously, and you may have several other dashboards or tools expecting to read its data from `<dbname>.<schemaname>.dim_customers`.
+
+You could use the `alias` configuration:
+
+<File name="models/schema.yml">
 
 ```yml
       - v: 1
-        defined_in: dim_customers # keep original relation name
+        config:
+          alias: dim_customers   # keep v1 in its original database location
 ```
 
+</File>
+
+Or, you could define a separate view that always points to the latest version of the model. We recommend this pattern because it's the most transparent and easiest to follow.
+
+<File name="models/dim_customers_view.yml">
+
+```sql
+{{ config(alias = 'dim_customers') }}
+
+select * from {{ ref('dim_customers') }}
+```
+
+</File>
```

Comment:
As I was building my table of examples up top, I decided that if we're not going to do this magically behind the scenes, I think we should just encourage people to move the alias definition around in their YAML as they progress their models. Telling people to shepherd an entire extra model around by hand feels gross.

Reply:
@joellabes I've had another thought here: It would be possible to implement this as a standard pattern, with a modification to the [`generate_alias_name`](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/get_custom_name/get_custom_alias.sql) macro.

```sql
{% macro generate_alias_name(custom_alias_name=none, node=none) -%}
    {%- if custom_alias_name -%}
        {{ return(custom_alias_name | trim) }}
    {%- elif node.version and not node.is_latest_version -%} {# <--- this bit #}
        {{ return(node.name ~ "_v" ~ (node.version | replace(".", "_"))) }}
    {%- else -%} {# latest version has standard behavior #}
        {{ return(node.name) }}
    {%- endif -%}
{%- endmacro %}
```

This way, whatever the `latest_version` is always lands in the canonical

**Should we make that the default behavior?** Or should it be something that end users opt into? I'm inclined to be a bit more opinionated, and say this should be the default—**really** emphasize that the latest version is _the thing_, and the old/new versions are mechanisms for managing change—but it does add a bit more inconsistency.

Reply:
Yes I definitely agree it should be the default. More to come on the Slack thread